src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -312,6 +312,14 @@ private async Task FormatDocumentCreatedFromTemplateAsync(IVsHierarchy hierarchy
             // We have to discover .editorconfig files ourselves to ensure that code style rules are followed.
             // Normally the project system would tell us about these.
             projectToAddTo = AddEditorConfigFiles(projectToAddTo, Path.GetDirectoryName(filePath));
+
+            // Because we're adding the initial project to the solution, we need to ensure that this solution snapshot
+            // has the right fallback analyzer options.  This normally happens in Workspace.SetCurrentSolutionAsync as
+            // it mutates.  But that has never happened so far, so we have to simulate that manually here.  This ensures
+            // we pick up the right host/vs options which is needed in order to run the code cleanup pass below.
+            solution = solution.WithFallbackAnalyzerOptions(Workspace.ComputeFinalFallbackAnalyzerOptions(solution, projectToAddTo.Solution));
```

Comment:
i don't love this.  alternatively, above this, we call solution.AddProject.  if we're comfortable with the idea, i could make it so that this logic automatically happens in .Add/.RemoveProject so that our in-memory forks reread this host data.

It feels OOGy to me though.  because that means you might have a solution snapshot with one set of options, and you fork it,a nd now the options are different due to the underlying host.  it feels like we don't want that, and options  in forked snapshots should not change unless explicit.  instead, only when the host updates .CurrentSolution should it change.

So this, while oogy, feels the cleanest to me.

Reply:
i suppose i could just do:

```
            var provider = oldSolution.Services.GetRequiredService<IFallbackAnalyzerConfigOptionsProvider>();
            newFallbackOptions = newFallbackOptions.Add(language, provider.GetOptions(language));
```

though the benefit of having and calling into the helper is that it works regardless of the state the workspace is in when FormatDocumentCreatedFromTemplateAsync is called.

Reply:
This does the same thing now, no?
```
private static Solution InitializeAnalyzerFallbackOptions(Solution oldSolution, Solution newSolution)
        => newSolution.WithFallbackAnalyzerOptions(ComputeFinalFallbackAnalyzerOptions(oldSolution, newSolution));
```